### PR TITLE
Add static assertion in engine that bool is no bigger than char

### PR DESCRIPTION
**Pull request by livecodeali**

### CODE REVIEW

File: engine/src/sysdefs.h [hunk 1]
```diff
@@ -138,26 +159,7 @@ along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
...
-#include <foundation-objc.h>
-#endif
-
-//////////////////////////////////////////////////////////////////////
-//
-//  FOUNDATION SYSTEM LIBRARY
-//
-
-#include <foundation-system.h>
+MCStaticAssert(sizeof(bool) <= sizeof(char));
```
Comment (runrevmark) [inline engine/src/sysdefs.h]: We can use static_assert directly now - I suggest:
```
// The engine is implemented assuming that 'bool' is at most one byte in size
static_assert(sizeof(bool) <= 1, "Bool size is not at most 1 byte")
```

Comment (livecodeali) [reply engine/src/sysdefs.h]: @runrevmark Done, thanks!

**Comment by runrevmark** [comment]
@livecode-vulcan review ok f2395bb

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok f2395bbc4ba650a0fbb7c5a087965b668c8ab053

**Comment by livecode-vulcan** [comment]
😞 test failure f2395bbc4ba650a0fbb7c5a087965b668c8ab053

* try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/1143)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/1145)
* try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/1138)
* try-community-emscripten: [pending](https://vulcan.livecode.com/builders/try-community-emscripten/builds/1144)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/1148)
* try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/1158)
* try-community-android-armv6: [pending](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/1139)

**Comment by livecodeali** [comment]
@livecode-vulcan retry f2395bb

**Comment by livecode-vulcan** [comment]
😎 test success f2395bbc4ba650a0fbb7c5a087965b668c8ab053

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/1144)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/1146)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/1139)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/1145)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/1149)
* try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/1159)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/1140)

**Pull request merged**